frontend/src/hooks/usePollenItems.js

```diff
@@ -1,19 +1,23 @@
-import { useEffect, useState } from 'react';
-import * as apiservice from '../service/apiservice';
+import { useContext, useEffect, useState } from 'react';
+import AuthContext from '../context/AuthContext';
+import axios from 'axios';
 
 export default function usePollenItems() {
   const [pollenItems, setPollenItems] = useState([]);
+  const { token } = useContext(AuthContext);
 
   useEffect(() => {
-    loadPollenData();
-  }, []);
-
-  const loadPollenData = () => {
-    apiservice
-      .loadAllPollenItems()
+    const config = {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    };
+    axios
+      .get(`/api/dictionary/`, config)
+      .then((response) => response.data)
       .then(setPollenItems)
-      .catch((err) => console.log(err.message));
-  };
+      .catch((error) => console.error(error.message));
+  }, [token]);
```

Comment:
you are using axios functions quite a lot in your react components - you may want to consider creating a service class in which you can include all your API requests where the token is retrieved once when needed 
this allows reduction of complexity in your components as well as pushing changes in your API authentication to a single class